client/jetpack-onboarding/steps/site-title.jsx

```diff
@@ -9,17 +9,63 @@ import { localize } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
+import Button from 'components/button';
+import Card from 'components/card';
 import FormattedHeader from 'components/formatted-header';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextarea from 'components/forms/form-textarea';
+import Main from 'components/main';
 
 class JetpackOnboardingSiteTitleStep extends React.PureComponent {
+	state = {
+		description: '',
+		title: '',
+	};
+
+	setDescription = event => {
+		this.setState( { description: event.target.value } );
+	};
+
+	setTitle = event => {
+		this.setState( { title: event.target.value } );
+	};
+
 	render() {
 		const { translate } = this.props;
 		const headerText = translate( "Let's get started." );
 		const subHeaderText = translate(
 			'First up, what would you like to name your site and have as its public description?'
 		);
 
-		return <FormattedHeader headerText={ headerText } subHeaderText={ subHeaderText } />;
+		return (
+			<Main>
+				<FormattedHeader headerText={ headerText } subHeaderText={ subHeaderText } />
+
+				<Card>
+					<form>
+						<FormFieldset>
+							<FormLabel htmlFor="title">{ translate( 'Site Title' ) }</FormLabel>
+							<FormTextInput id="title" onChange={ this.setTitle } value={ this.state.title } />
+						</FormFieldset>
+
+						<FormFieldset>
+							<FormLabel htmlFor="description">{ translate( 'Site Description' ) }</FormLabel>
+							<FormTextarea
+								id="description"
+								onChange={ this.setDescription }
+								value={ this.state.description }
+							/>
+						</FormFieldset>
+
+						<Button href={ this.props.getForwardUrl() } primary>
```

Comment:
The `getForwardUrl` is quite useful, isn't it? 😉 

But seriously, how about updating the `Button` to a `FormButton`, and adding a submit handler to the form (and moving the redirect part to handler)? This would allow the user to submit the form by other means (for example, by pressing Return/Enter while focusing one of the fields), and would later allow us to quickly update the component to store the fields in Redux or send them to the server.

Reply:
Yeah, can do. `FormButton`'s label is hard-wired to read 'Save Settings' tho (and 'Saving...' while it is, well, saving).

> and would later allow us to quickly update the component to store the fields in Redux or send them to the server.

JTFR, that would be possible with the `Button` too, tho: by keeping the `href` and additionally adding an `onClick` handler :slightly_smiling_face:

Reply:
> Yeah, can do. `FormButton`'s label is hard-wired to read 'Save Settings' tho (and 'Saving...' while it is, well, saving).

Oh, I just noticed I was mistaken here -- it _is_ possible to customize that label.

> JTFR, that would be possible with the `Button` too, tho: by keeping the `href` and additionally adding an `onClick` handler :slightly_smiling_face: 

To elaborate a bit more, I generally like to avoid `page()` redirects in action handlers, if there is any option that allows me to set an `href` or similar.

Reply:
Hmm, I had another look at the [MDN docs for `button`](https://developer.mozilla.org/en/docs/Web/HTML/Element/button). I might be inclined to stick with `Button` for now, mostly for the aforementioned reason, plus:

> `type`
> The type of the button. Possible values are:
> `submit`: The button submits the form data to the server. This is the default if the attribute is not specified, or if the attribute is dynamically changed to an empty or invalid value.
> reset: The button resets all the controls to their initial values.
> `button`: The button has no default behavior. It can have client-side scripts associated with the element's events, which are triggered when the events occur.

Since we're doing client-side processing here, sounds like `submit` isn't quite the right `type`...

Reply:
> To elaborate a bit more, I generally like to avoid page() redirects in action handlers, if there is any option that allows me to set an href or similar.

Yeah, makes sense to avoid `page` redirects if possible, but I really hope it won't compromise the usability. I'm happy with any approach if users are able to submit the form in a native way using `Return/Enter` while focusing one of the fields.

Reply:
> we're doing client-side processing here

Let's be fair, we're not quite sure about this yet 😉

Reply:
> Yeah, makes sense to avoid `page` redirects if possible, but I really hope it won't compromise the usability. I'm happy with any approach if users are able to submit the form in a native way using `Return/Enter` while focusing one of the fields. 

Fair enough, but the Description field is multiline 😛

Reply:
> the Description field is multiline

😆 

Well, the title field isn't so ¯\\_(ツ)_/¯